website/docs/plugin/framework/handling-data/blocks.mdx

```diff
@@ -0,0 +1,31 @@
+---
+page_title: 'Plugin Development - Framework: Handling Data - Blocks'
+description: >-
+  Blocks.
+---
+
+# Blocks
+
+A block is a container for other content:
+
+```hcl
+resource "aws_instance" "example" {
+  ami = "abc123"
+
+  network_interface {
+    # ...
+  }
+}
+```
+
+A block has a type (resource in this example). Each block type defines how many labels must follow the type keyword. The resource block type expects two labels, which are aws_instance and example in the example above. A particular block type may have any number of required labels, or it may require none as with the nested network_interface block type.
+
+After the block type keyword and any labels, the block body is delimited by the { and } characters. Within the block body, further arguments and blocks may be nested, creating a hierarchy of blocks and their associated arguments.
+
+The Terraform language uses a limited number of top-level block types, which are blocks that can appear outside of any other block in a configuration file. Most of Terraform's features (including resources, input variables, output values, data sources, etc.) are implemented as top-level blocks.
+
+## List, Set and Single Nested Blocks
+
+Wherever possible, the usage of [nested attributes](/plugin/framework/handling-data/terraform-concepts#nested-attributes) is preferred.
+
+The three options for [nested blocks](/plugin/framework/handling-data/terraform-concepts#nested-blocks) are `ListNestedBlock`, `SetNestedBlock` and `SingleNestedBlock`.
```

Comment:
```suggestion
The configuration syntax for a provider-defined block is a type (block name) and the body is delimited by the `{` and `}` characters. Each block has an associated nesting mode, which declares whether a block is repeatable. The available nesting modes are:

- List: Ordered collection of objects
- Set: Unordered collection of objects
- Single: One object

For list and set blocks, configurations can implement [`dynamic` block expressions](/language/expressions/dynamic-blocks) to reduce hardcoded block values.

Refer to [nested blocks](/plugin/framework/handling-data/terraform-concepts#nested-blocks) for implementation examples in framework code.
```

Reply:
Note: it may be worth putting the block code examples on this page, rather than the concepts page.